presto-main/src/test/java/io/prestosql/sql/planner/TestPredicatePushdown.java

```diff
@@ -111,7 +111,7 @@ public void testNonStraddlingJoinExpression()
                                 anyTree(
                                         tableScan("orders", ImmutableMap.of("ORDERS_OK", "orderkey"))),
                                 anyTree(
-                                        filter("cast('2' as varchar) = cast(LINEITEM_LINENUMBER as varchar)",
+                                        filter("NOT(LINEITEM_OK IS NULL) AND cast('2' as varchar) = cast(LINEITEM_LINENUMBER as varchar)",
```

Comment:
After PPD run it's OK to have such redundant filter conditions, as long as they are subsequently eliminated.
(I think `testPushDownJoinConditionConjunctsToInnerSideBasedOnInheritedPredicate` covers that)

Please update expected plans in `io.prestosql.sql.planner.TestLogicalPlanner`, especially `io.prestosql.sql.planner.TestLogicalPlanner#testJoin` -- assert there is a `filter` and check filter's condition.

Reply:
For plain expressions the redundant expressions are eliminated but I guess we can extend the same for casted expression also.

Reply:
I think this could be a follow up PR